plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java

```diff
@@ -687,10 +720,10 @@ public void prepareNextOutputFile() throws KettleException {
       // file is guaranteed to be in place now
       try ( InputStream inputStream = KettleVFS.getInputStream( data.file ) ) {
         if ( XLSX.equalsIgnoreCase( meta.getExtension() ) ) {
-          // Ignore, by now, if it's to use streaming!
-          // In that case, one needs to initialize it later, after writing header/template, because
-          // SXSSFWorkbook can't read/rewrite existing data, only append.
-          data.wb = new XSSFWorkbook( inputStream );
+          Workbook workbook = WorkbookFactory.create(inputStream);
+          SXSSFWorkbook sxssfWorkbook = new SXSSFWorkbook();
+          copyData(workbook, sxssfWorkbook);
+          data.wb = sxssfWorkbook;
```

Comment:
This is selecting streaming mode even if `meta.isStreamingData()` (`Stream XLSX Data` option) is set to false.
If you want to remove that option, you need to also remove it from the UI. This shouldn't be done in a bug, and you should seek PM approval first.

Reply:
I dont want to remove that option, just updated the code am checking the meta.isStreamingData condition if we want to create the SXSSFWorkbook instance

Reply:
So now what is the improvement that these changes bring, and how do they fix BACKLOG-41001?

Reply:
BTW I made a comment in the jira ticket, don't know if it was noticed

Reply:
previously by default we use to create SXSSFWorkbook, when we are dealing with huge amount of data we used to get outofmemory exception, so from now we with the following check meta.isStreamingData(), we will be avoid that by creating the XSSFWorkbook 
If we are crossing this limit  (1,048,575 rows and 16,384 columns), we need to split the excel sheet by rows .

Reply:
When in streaming mode there wouldn't be an OOM error before. And with it disabled there will still be one after this change.

Reply:
coding wise I am not disabling the streaming mode option, with the steaming mode condition I am just creating the SXSSFWorkbook instance which we were not doing before , with the existing code I have seen like XSSFWorkbook is being created by default which is causing OOM exception

Reply:
> I am just creating the SXSSFWorkbook instance which we were not doing before

We were already doing that a bit below:
https://github.com/pentaho/pentaho-kettle/blob/200f687aa127113aad7a49f2dbab97ae5172459a/plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java#L752

If you test current 10.2 with the streaming option enabled, there will be no OoM error.

Reply:
I believe the issue should be reevaluated, taking the streaming option into account.

Reply:
yeap its working fine with streaming option

Reply:
Hi, we have rolledback changes related to streaming version workbook, going with changes related to upgrading the apache poi

Reply:
The commit msg still mentions the other issue, can you change that please?